Applications/Spire/Include/Spire/Ui/SecurityView.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef SPIRE_SECURITY_VIEW_HPP
+#define SPIRE_SECURITY_VIEW_HPP
+#include <QStackedWidget>
+#include "Nexus/Definitions/Security.hpp"
+#include "Spire/Ui/ComboBox.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /**
+   * Displays securities and provides the functionality to quickly search and
+   * cycle through securities.
+   */
+  class SecurityView : public QWidget {
+    public:
+
+      /** A ValueModel over a Nexus::Security. */
+      using CurrentModel = ValueModel<Nexus::Security>;
+
+      /**
+       * Constructs a SecurityView using a default local model.
+       * @param query_model The model used to query matches.
+       * @param body The component that represents the current security.
+       * @param parent The parent widget.
+       */
+      SecurityView(std::shared_ptr<ComboBox::QueryModel> query_model,
+        QWidget* body, QWidget* parent = nullptr);
+
+      /**
+       * Constructs a SecurityView.
+       * @param query_model The model used to query matches.
+       * @param body The component that represents the current security.
+       * @param current The current value's model.
+       * @param parent The parent widget.
+       */
+      SecurityView(std::shared_ptr<ComboBox::QueryModel> query_model,
+        QWidget* body, std::shared_ptr<CurrentModel> current,
```

Comment:
Write your code in a consistent order. The constructor orders things as ```query_model```, ```body```, ```current```, then the methods order things as ```query_model```, ```current```, ```body```, and then the fields order things as ```body```, ```query_model```, ```current```.

Determine what the conceptual order of these fields are and use a consistent ordering.

Reply:
The ordering is still not consistent, the constructor is now ```query_model```, ```body```, ```current``` and the methods are ```query_model```, ```current```, ```body```. These subtle details add up in the long run. Think about conceptually what the order of dependencies are and order things from the most independent to the least independent. What is more likely, that ```current``` depends on ```query_model``` or that ```query_model``` depends on ```current```? Is it more likely that ```body``` depends on ```current``` or that ```current``` depends on ```body```?

Once you know what the most common logical order of dependencies are, arrange everything in order from independent to dependent, constructor, methods, fields, algorithms, anything and everything.